launch/launch/actions/group_action.py

```diff
@@ -65,24 +85,40 @@ def parse(cls, entity: Entity, parser: Parser):
         """Return `GroupAction` action and kwargs for constructing it."""
         _, kwargs = super().parse(entity, parser)
         scoped = entity.get_attr('scoped', data_type=bool, optional=True)
+        forwarding = entity.get_attr('forwarding', data_type=bool, optional=True)
         if scoped is not None:
             kwargs['scoped'] = scoped
+        if forwarding is not None:
+            kwargs['forwarding'] = forwarding
         kwargs['actions'] = [parser.parse_action(e) for e in entity.children]
         return cls, kwargs
 
     def get_sub_entities(self) -> List[LaunchDescriptionEntity]:
         """Return subentities."""
         if self.__actions_to_return is None:
-            self.__actions_to_return = []  # type: List[Action]
-            self.__actions_to_return += [
+            self.__actions_to_return = list(self.__actions)
+            configuration_sets = [
                 SetLaunchConfiguration(k, v) for k, v in self.__launch_configurations.items()
             ]
-            self.__actions_to_return += list(self.__actions)
             if self.__scoped:
+                if self.__forwarding:
+                    self.__actions_to_return = [
+                        PushLaunchConfigurations(),
+                        *configuration_sets,
+                        *self.__actions_to_return,
+                        PopLaunchConfigurations()
+                    ]
+                else:
+                    self.__actions_to_return = [
+                        PushLaunchConfigurations(),
+                        ResetLaunchConfigurations(self.__launch_configurations),
+                        *self.__actions_to_return,
+                        PopLaunchConfigurations()
+                    ]
+            else:
                 self.__actions_to_return = [
-                    PushLaunchConfigurations(),
-                    *self.__actions_to_return,
-                    PopLaunchConfigurations()
+                    *configuration_sets,
+                    *self.__actions_to_return
```

Comment:
nitpick: this logic might be reduced by keeping the:

```python
if self.__scoped:
    self.__actions_to_return = [
         PushLaunchConfigurations(),
         *self.__actions_to_return,
         PopLaunchConfigurations(),
    ]
```

And doing that unconditionally, rather than doing it in both the cases where forwarding is true or false. Dunno if that made sense... something like this:

```python
if self.__forwarding:
    self.__actions_to_return = [
        *configuration_sets,
        *self.__actions_to_return,
    ]
else:
    self.__actions_to_return = [
        ResetLaunchConfigurations(self.__launch_configurations),
        *self.__actions_to_return,
    ]

if self.__scoped:
    self.__actions_to_return = [
        PushLaunchConfigurations(),
        *self.__actions_to_return,
        PopLaunchConfigurations(),
    ]
```

Reply:
@wjwwood I am ok with that and like the idea of cleaning up the logic. It does expose the case of `forwarding=False` and `scoped=False` which was not possible with the previous implementation. The end result would be clearing launch configurations in the current scope which will clear launch configurations for anything that comes after the group action. Is that a use case that should be exposed to users? If so your suggestion works well and I can add the documentation for it. 

Thoughts?

Reply:
I think that scoped needs to be true for anything in forwarding to be done, so maybe the logic needs to be nested as it is, or the else case for forwarding needs to be `elif scoped`.

Reply:
That is to say, I don't think anything should happen when both scoped and forwarding are false.

Let me know what you will do here, either change it or leave it as-is, so I know when it's ready for another review.

Reply:
@wjwwood I think I am going to leave it as is. I have tried a couple different versions and they are not much cleaner than the existing. Besides documentation, it could be mitigated with either printing a warning or raising an error if scoped and forwarding are both False. Printing a warning creates a dependency on `launch.logging` but allows execution to continue, raising an error would stop execution. Thoughts?

    def execute(self, context: LaunchContext) -> Optional[List[LaunchDescriptionEntity]]:
        """Execute the action."""
        if self.__scoped is False and self.__forwarding is False:
            launch.logging.get_logger('GroupAction').warn(
                'forwarding = False will be ignored when scoped = False.')
            
            # or

            raise RuntimeError('GroupAction cannot have forwarding = False while scoped = False')

        return self.get_sub_entities()